x/ecocredit/server/core/query_batches.go

```diff
@@ -0,0 +1,45 @@
+package core
+
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+// Batches queries for all batches in the given credit class.
+func (k Keeper) Batches(ctx context.Context, request *v1beta1.QueryBatchesRequest) (*v1beta1.QueryBatchesResponse, error) {
+	pg, err := GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+	project, err := k.stateStore.ProjectInfoStore().GetByName(ctx, request.ProjectId)
+	if err != nil {
+		return nil, err
+	}
+	it, err := k.stateStore.BatchInfoStore().List(ctx, ecocreditv1beta1.BatchInfoProjectIdIndexKey{}.WithProjectId(project.Id), ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We only need the id from project, which we already have in the request. Do we need to get the project from state or can we just convert the id in the request?

Reply:
nvm